frontend/src/concepts/pipelines/content/pipelineSelector/PipelineVersionSelector.tsx

```diff
@@ -0,0 +1,168 @@
+import {
+  EmptyStateVariant,
+  HelperText,
+  HelperTextItem,
+  Icon,
+  Menu,
+  MenuContainer,
+  MenuContent,
+  MenuList,
+  MenuSearch,
+  MenuSearchInput,
+  MenuToggle,
+  SearchInput,
+  Spinner,
+} from '@patternfly/react-core';
+import { TableVariant } from '@patternfly/react-table';
+import PipelineSelectorTableRow from '~/concepts/pipelines/content/pipelineSelector/PipelineSelectorTableRow';
+import { TableBase } from '~/components/table';
+import { usePipelineVersionLoadMore } from '~/concepts/pipelines/content/tables/usePipelineLoadMore';
+import { PipelineVersionKF } from '~/concepts/pipelines/kfTypes';
+import { pipelineVersionSelectorColumns } from '~/concepts/pipelines/content/pipelineSelector/columns';
+import usePipelineVersionsTable from '~/concepts/pipelines/content/tables/pipelineVersion/usePipelineVersionsTable';
+import PipelineViewMoreFooterRow from '~/concepts/pipelines/content/tables/PipelineViewMoreFooterRow';
+import {
+  getSelectorTableColumnSort,
+  useSelectorSearch,
+} from '~/concepts/pipelines/content/pipelineSelector/utils';
+import EmptyTableView from '~/concepts/pipelines/content/tables/EmptyTableView';
+
+type PipelineVersionSelectorProps = {
+  pipelineId?: string;
+  selection?: string;
+  onSelect: (version: PipelineVersionKF) => void;
+};
+
+const PipelineVersionSelector: React.FC<PipelineVersionSelectorProps> = ({
+  pipelineId,
+  selection,
+  onSelect,
+}) => {
+  const [isOpen, setOpen] = React.useState(false);
+
+  const [
+    [{ items: initialData, totalSize: fetchedSize, nextPageToken: initialPageToken }, loaded],
+    { initialLoaded, ...tableProps },
+  ] = usePipelineVersionsTable(pipelineId)();
+
+  const toggleRef = React.useRef(null);
+  const menuRef = React.useRef(null);
+
+  const { sortDirection, sortField, setFilter, filter } = tableProps;
+  const { data: versions, onLoadMore } = usePipelineVersionLoadMore({
+    pipelineId,
+    initialData,
+    initialPageToken,
+    sortDirection,
+    sortField,
+    filter,
+    loaded,
+  });
+
+  const { totalSize, ...searchProps } = useSelectorSearch({ setFilter, fetchedSize, loaded });
+  const { onClear } = searchProps;
+
+  const menu = (
+    <Menu data-id="pipeline-version-selector-menu" ref={menuRef} isScrollable>
+      <MenuContent>
+        <MenuSearch>
+          <MenuSearchInput>
+            <SearchInput {...searchProps} aria-label="Filter pipeline versions" />
+          </MenuSearchInput>
+          <HelperText>
+            <HelperTextItem variant="indeterminate">{`Type a name to search your ${totalSize} versions.`}</HelperTextItem>
+          </HelperText>
+        </MenuSearch>
+        <MenuList>
+          <div role="menuitem">
+            <TableBase
+              {...tableProps}
+              itemCount={fetchedSize}
+              loading={!loaded}
+              data-id="pipeline-version-selector-table-list"
+              emptyTableView={
+                <EmptyTableView
+                  hasIcon={false}
```

Comment:
```suggestion
```

Reply:
Why do we want to remove this? I figured out that the empty state in the selector could be very tiny so we might not want an icon for it. (Followed the extra small empty state example here https://www.patternfly.org/components/empty-state#extra-small)

Reply:
@DaoDaoNoCode oh i thought that `hasIcon = false` was the default is all. If it's not the default then certainly keep it.